CONTRIBUTING.md

```diff
@@ -28,11 +28,46 @@ Tests have been validated using:
 They may work with older versions, though.
 
 ### Running the Tests
-Go sources are located under the `test` directory. The repository uses [Go workspace][go-ws], so tests can be run from the repository root using following command:
+Go sources are located under the `test` directory. The repository uses [Go workspace][go-ws], so tests can be run from the repository root:
+
+#### Unit Tests
 
 ```shell
  make test
  ```
+
+#### End-to-End Tests
+[Pulumi](https://www.pulumi.com/) is used to deploy End-to-End infrastructures, defined as "stacks", using the [`test-infra-definitions`](https://github.com/DataDog/test-infra-definitions) and [`datadog-agent`](https://github.com/DataDog/datadog-agent/tree/main/test/new-e2e) E2E frameworks.
```

Comment:
```suggestion
[Pulumi](https://www.pulumi.com/) is used to deploy infrastructure end-to-end (E2E), defined as "stacks", using the [`test-infra-definitions`](https://github.com/DataDog/test-infra-definitions) and [`datadog-agent`](https://github.com/DataDog/datadog-agent/tree/main/test/new-e2e) E2E frameworks.
```

Reply:
@buraizu I'd like to keep the phrase, "end-to-end infrastructures", because `end-to-end` in this case refers to the "end-to-end" type of testing methodology. Perhaps I can change it to say "end-to-end testing infrastructures". What do you think?

```suggestion
[Pulumi](https://www.pulumi.com/) is used to deploy end-to-end testing infrastructures, defined as "stacks", using the [`test-infra-definitions`](https://github.com/DataDog/test-infra-definitions) and [`datadog-agent`](https://github.com/DataDog/datadog-agent/tree/main/test/new-e2e) E2E frameworks.
```

Reply:
@fanny-jiang thanks for following up. Initially I thought `infrastructure` had to be used as a singular noun, but did a little more looking into it and it's probably fine in this context. I updated the suggestion accordingly and approved, thanks again!

Reply:
Thanks!